<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Meta-eukomic</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Meta-eukomic</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./main.html" rel="" target="" aria-current="page">
 <span class="menu-text">Main</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#meta-eukomic" id="toc-meta-eukomic" class="nav-link active" data-scroll-target="#meta-eukomic"><span class="header-section-number">1</span> Meta-eukomic</a>
  <ul class="collapse">
  <li><a href="#background-reading" id="toc-background-reading" class="nav-link" data-scroll-target="#background-reading"><span class="header-section-number">1.1</span> Background reading</a></li>
  <li><a href="#working-environment" id="toc-working-environment" class="nav-link" data-scroll-target="#working-environment"><span class="header-section-number">1.2</span> Working environment</a></li>
  </ul></li>
  <li><a href="#step-by-step" id="toc-step-by-step" class="nav-link" data-scroll-target="#step-by-step"><span class="header-section-number">2</span> Step-by-step</a>
  <ul class="collapse">
  <li><a href="#determine-assembly-groups" id="toc-determine-assembly-groups" class="nav-link" data-scroll-target="#determine-assembly-groups"><span class="header-section-number">2.1</span> 1. Determine assembly groups</a></li>
  <li><a href="#trim-qc" id="toc-trim-qc" class="nav-link" data-scroll-target="#trim-qc"><span class="header-section-number">2.2</span> 2. Trim &amp; QC</a></li>
  <li><a href="#remove-ribosomal-rna" id="toc-remove-ribosomal-rna" class="nav-link" data-scroll-target="#remove-ribosomal-rna"><span class="header-section-number">2.3</span> 3. Remove ribosomal RNA</a>
  <ul class="collapse">
  <li><a href="#run-sortmerna" id="toc-run-sortmerna" class="nav-link" data-scroll-target="#run-sortmerna">Run SortMeRNA</a></li>
  </ul></li>
  <li><a href="#assembly" id="toc-assembly" class="nav-link" data-scroll-target="#assembly"><span class="header-section-number">2.4</span> 4. Assembly</a>
  <ul class="collapse">
  <li><a href="#megahit-assembly-1" id="toc-megahit-assembly-1" class="nav-link" data-scroll-target="#megahit-assembly-1">MEGAHIT (assembly 1)</a></li>
  <li><a href="#idba-tran" id="toc-idba-tran" class="nav-link" data-scroll-target="#idba-tran">IDBA-Tran</a></li>
  </ul></li>
  <li><a href="#evaluate-assemblies" id="toc-evaluate-assemblies" class="nav-link" data-scroll-target="#evaluate-assemblies"><span class="header-section-number">2.5</span> 5. Evaluate assemblies</a>
  <ul class="collapse">
  <li><a href="#quast" id="toc-quast" class="nav-link" data-scroll-target="#quast">QUAST</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a></li>
  </ul></li>
  <li><a href="#metatranscriptome-run-information" id="toc-metatranscriptome-run-information" class="nav-link" data-scroll-target="#metatranscriptome-run-information"><span class="header-section-number">2.6</span> Metatranscriptome run information</a></li>
  </ul></li>
  <li><a href="#eukrhythmic" id="toc-eukrhythmic" class="nav-link" data-scroll-target="#eukrhythmic"><span class="header-section-number">3</span> eukrhythmic</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Meta-eukomic</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="meta-eukomic" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Meta-eukomic</h1>
<p><em>SKH</em> last updated June 2024</p>
<p>Location for raw files: <code>/scratch/group/hu-lab/meta-eukomics/raw-data</code> Individual R1 and R2 files are 2.9 and 2.7 G, respectively.</p>
<section id="background-reading" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="background-reading"><span class="header-section-number">1.1</span> Background reading</h2>
<ol type="1">
<li><p>Cohen NR, Alexander H, Krinos AI, Hu SK, Lampe RH. Marine Microeukaryote Metatranscriptomics: Sample Processing and Bioinformatic Workflow Recommendations for Ecological Applications. Frontiers in Marine Science 2022; 9.</p></li>
<li><p>Krinos AI, Cohen NR, Follows MJ, Alexander H. Reverse engineering environmental metatranscriptomes clarifies best practices for eukaryotic assembly. BMC Bioinformatics 2023; 24: 74.</p></li>
</ol>
</section>
<section id="working-environment" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="working-environment"><span class="header-section-number">1.2</span> Working environment</h2>
<p>High performace computer (HPC) hosted by my University. This is the TAMU HPRC</p>
</section>
</section>
<section id="step-by-step" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Step-by-step</h1>
<section id="determine-assembly-groups" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="determine-assembly-groups"><span class="header-section-number">2.1</span> 1. Determine assembly groups</h2>
</section>
<section id="trim-qc" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="trim-qc"><span class="header-section-number">2.2</span> 2. Trim &amp; QC</h2>
<p><em>Initial Fastqc</em></p>
<p>Code for slurm script run on the HPC to run fastqc (<code>fastqc.slurm</code>). For this set of samples, it took 10 minutes and used 290 MB. CPU used: 00:12:27.</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>module load FastQC<span class="sc">/</span><span class="dv">0</span>.<span class="fl">11.9</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb1-2"><a href="#cb1-2"></a></span>
<span id="cb1-3"><a href="#cb1-3"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001.fastq.gz</span>
<span id="cb1-4"><a href="#cb1-4"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_001.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In order to look at the output <code>.html</code> files, they need to be opened locally.</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>scp <span class="sc">$</span>HPRC<span class="sc">-</span>ADDRESS<span class="sc">:</span><span class="er">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001_fastqc.html meta<span class="sc">-</span>eukomic<span class="sc">/</span>output<span class="sc">-</span>files<span class="sc">/</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then we need to trim the individual reads, removing any sequencing-based primers, etc. We can use the program, <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a> for this. R1 reads are forward and R2 are reverse. The Trimmomatic software requires input reads and then output trimmed and unpaired reads (the latter of which are discarded). Another input file required is a list of the possible primers and adapters from sequencing, <code>adapters-primers.fa</code>.</p>
<p>Slurm script, <code>trim_fastqc.slurm</code>. Below, trim parameter include:</p>
<ul>
<li><p>Remove adapters, found in <code>adapters-primers.fa</code></p></li>
<li><p>Remove leading low quality or N bases (below quality 3) (LEADING:3)</p></li>
<li><p>Remove trailing low quality or N bases (below quality 3) (TRAILING:3)</p></li>
<li><p>Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 10 (SLIDINGWINDOW:4:10)</p></li>
<li><p>Drop reads shorter than 50 bases long (MINLEN:50)</p></li>
</ul>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>module load Trimmomatic<span class="sc">/</span><span class="fl">0.39</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a>java <span class="sc">-</span>jar <span class="sc">$</span>EBROOTTRIMMOMATIC<span class="sc">/</span>trimmomatic<span class="dv">-0</span>.<span class="fl">39.</span>jar PE <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_001.fastq.gz  <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_unpaired.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_unpaired.fastq.gz ILLUMINACLIP<span class="sc">:</span>adapters<span class="sc">-</span>primers.fa<span class="sc">:</span><span class="dv">2</span><span class="sc">:</span><span class="dv">30</span><span class="sc">:</span><span class="dv">10</span> LEADING<span class="sc">:</span><span class="dv">3</span> TRAILING<span class="sc">:</span><span class="dv">3</span> SLIDINGWINDOW<span class="sc">:</span><span class="dv">4</span><span class="sc">:</span><span class="dv">10</span> MINLEN<span class="sc">:</span><span class="dv">50</span></span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># "$EBROOTTRIMMOMATIC/trimmomatic-0.39.jar" note that this is specific to the HPRC system we are using</span></span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a>echo <span class="st">"Trimmomtatic complete. Repeating fastqc"</span></span>
<span id="cb3-8"><a href="#cb3-8"></a></span>
<span id="cb3-9"><a href="#cb3-9"></a>module load FastQC<span class="sc">/</span><span class="dv">0</span>.<span class="fl">11.9</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb3-10"><a href="#cb3-10"></a></span>
<span id="cb3-11"><a href="#cb3-11"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz</span>
<span id="cb3-12"><a href="#cb3-12"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="remove-ribosomal-rna" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="remove-ribosomal-rna"><span class="header-section-number">2.3</span> 3. Remove ribosomal RNA</h2>
<p><a href="https://sortmerna.readthedocs.io/en/latest/installation.html">Sortmerna</a> installation guidelines and usage can be found here.</p>
<p>On the TAMU HPRC, this is where we needed to make a conda environment to run this.</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># Load anaconda</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>module load Anaconda3<span class="sc">/</span><span class="fl">2022.05</span></span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># modify bioconda config for sortmeRNA</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>conda config <span class="sc">--</span>add channels defaults</span>
<span id="cb4-6"><a href="#cb4-6"></a>conda config <span class="sc">--</span>add channels bioconda</span>
<span id="cb4-7"><a href="#cb4-7"></a>conda config <span class="sc">--</span>add channels conda<span class="sc">-</span>forge</span>
<span id="cb4-8"><a href="#cb4-8"></a>conda config <span class="sc">--</span>set channel_priority strict</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Search for sortmerna using conda:</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>conda search sortmerna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Create a conda environment specific for sortmerna, activate it, and install it.</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>conda create <span class="sc">--</span>name sortmerna_env</span>
<span id="cb6-2"><a href="#cb6-2"></a>conda activate sortmerna_env</span>
<span id="cb6-3"><a href="#cb6-3"></a>conda install sortmerna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Answer “yes” (<em>Y</em>) to installation questions. Now when you run SortMeRNA, you need to activate this conda environment and execute your code. The next step is to download the databases to use for this program.</p>
<section id="sortmerna-databases" class="level4">
<h4 class="anchored" data-anchor-id="sortmerna-databases">SortMeRNA databases</h4>
<p>Separately, you need to download the databases to “align” and check the sequences for ribosomal RNAs.</p>
<p>Instructions for <a href="https://sortmerna.readthedocs.io/en/latest/databases.html">downloading version 4</a>:</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># Move to the location you want your rRNA databases stored</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>cd <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span> </span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co"># Download from github release </span></span>
<span id="cb7-5"><a href="#cb7-5"></a>wget https<span class="sc">:</span><span class="er">//</span>github.com<span class="sc">/</span>biocore<span class="sc">/</span>sortmerna<span class="sc">/</span>releases<span class="sc">/</span>download<span class="sc">/</span>v4.<span class="fl">3.4</span><span class="sc">/</span>database.tar.gz</span>
<span id="cb7-6"><a href="#cb7-6"></a></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co"># Make new directory for this and un-zip downloaded file</span></span>
<span id="cb7-8"><a href="#cb7-8"></a>mkdir rRNA_databases_v4</span>
<span id="cb7-9"><a href="#cb7-9"></a>tar <span class="sc">-</span>xvf database.tar.gz <span class="sc">-</span>C rRNA_databases_v4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-sortmerna" class="level3">
<h3 class="anchored" data-anchor-id="run-sortmerna">Run SortMeRNA</h3>
<p>In your slurm script, activate conda, and then the sortmerna environment:</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># Load anaconda</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>module load Anaconda3<span class="sc">/</span><span class="fl">2022.05</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>conda activate sortmerna_env</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a>mkdir <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA<span class="sc">-</span>sort</span>
<span id="cb8-6"><a href="#cb8-6"></a></span>
<span id="cb8-7"><a href="#cb8-7"></a>sortmerna <span class="sc">-</span>ref <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA_databases_v4<span class="sc">/</span>smr_v4<span class="fl">.3</span>_sensitive_db.fasta \</span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="sc">-</span>reads <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz \</span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="sc">-</span>reads <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="sc">-</span>workdir <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA<span class="sc">-</span>sort<span class="sc">/</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="sc">-</span>fastx <span class="sc">--</span>paired_in <span class="sc">--</span>out2 <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><code>paired_in</code>: The option ‘paired_in’ is optimal for users who want all reads in the other.fasta file to be non-rRNA. However, there are small chances that reads which are non-rRNA will also be put into the aligned.fasta file.</p>
<p>Since the SortmeRNA tool is optimitized to take things that are aligned with the rRNA databases and put them in a file called “aligned”, then we need to use the <code>other.fasta</code> output file.</p>
<p>How do I properly run sortmerna? https://github.com/sortmerna/sortmerna/issues/333</p>
</section>
</section>
<section id="assembly" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="assembly"><span class="header-section-number">2.4</span> 4. Assembly</h2>
<p>The next step is to take all the sequenced reads and bring them together to make longer, more continuous sequences, called <em>contigs</em>. Here, we will use two assemblers and combine the results. Each one is built slightly differently.</p>
<section id="megahit-assembly-1" class="level3">
<h3 class="anchored" data-anchor-id="megahit-assembly-1">MEGAHIT (assembly 1)</h3>
<p>We will first use <a href="https://github.com/voutcn/megahit">megahit</a>. To save the assemblies separately, make a new assembly output file in your working <code>scratch</code> space. <code>mkdir /scratch/group/hu-lab/meta-eukomics/assembly</code></p>
<p>The megahit command below, outputs contigs in the new <code>assembly</code> directory, only keeps reads longer than 100 bps, and uses the megahit preset for lots of diversity in a sample (<em>meta-large</em>).</p>
<p>Megahit uses multiple k-mer strategy, and we can set the min and max k. In order to reduce the complexity of the <em>de Bruijin</em> graph, a kmin size of 25 and higher (like 27) is better.</p>
<p>See slurm script: <code>megahit-assembly.slurm</code></p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>module load GCCcore<span class="sc">/</span><span class="dv">11</span>.<span class="fl">2.0</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>module load MEGAHIT<span class="sc">/</span><span class="dv">1</span>.<span class="fl">2.9</span></span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a>megahit <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span> <span class="sc">--</span>min<span class="sc">-</span>contig<span class="sc">-</span>len <span class="dv">1000</span> <span class="sc">--</span>presets meta<span class="sc">-</span>large <span class="sc">--</span>num<span class="sc">-</span>cpu<span class="sc">-</span>threads <span class="dv">32</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>With the above settings, megahit recovered 31,331 contigs, total 44610184 bp, min 1000 bp, max 10674 bp, avg 1423 bp, N50 1391 bp. And this took about <em>10 hours</em>.</p>
<p>On the HPC, this was run on 1 node, 4 cores per node, and it used 18.5 GBs of memory.</p>
<p>Megahit assembly location: <code>/scratch/group/hu-lab/meta-eukomics/assembly/megahit-output/final.contigs.fa</code></p>
<p><em><em>Optional</em>: Visualize your megahit assembly</em></p>
<p>When megahit assembly is complete, use this option to visualize it: https://github.com/voutcn/megahit/wiki/Visualizing-MEGAHIT’s-contig-graph</p>
</section>
<section id="idba-tran" class="level3">
<h3 class="anchored" data-anchor-id="idba-tran">IDBA-Tran</h3>
<p><a href="https://i.cs.hku.hk/~alse/hkubrg/projects/idba_tran/index.html">IDBA-tran</a> is another <em>de novo</em> assembler. Uses local assembly to reconstruct missing kmers in low-expressed transcripts.</p>
<p>See script: <code>idba-assembly.slurm</code> to run assembly with minimum kmer at 20 and max kmer at 50 with a 5 step increment of kmer.</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>echo <span class="st">"unzip files"</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>gunzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R<span class="sc">*</span>_trimmed.fastq.gz</span>
<span id="cb10-3"><a href="#cb10-3"></a></span>
<span id="cb10-4"><a href="#cb10-4"></a>module load GCC<span class="sc">/</span><span class="dv">8</span>.<span class="fl">2.0</span><span class="dv">-2</span>.<span class="fl">31.1</span> </span>
<span id="cb10-5"><a href="#cb10-5"></a>module load IDBA<span class="sc">-</span>UD<span class="sc">/</span><span class="dv">1</span>.<span class="fl">1.3</span></span>
<span id="cb10-6"><a href="#cb10-6"></a></span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co"># Convert fastq R1 and R2 files into a single 'interleaved` fastq files</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>fq2fa <span class="sc">--</span>merge <span class="sc">--</span>filter <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>merged<span class="sc">-</span>PE<span class="sc">-</span><span class="cf">for</span><span class="sc">-</span>idba.fa</span>
<span id="cb10-9"><a href="#cb10-9"></a></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="co"># Run assembly</span></span>
<span id="cb10-11"><a href="#cb10-11"></a>idba_tran <span class="sc">-</span>r <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>merged<span class="sc">-</span>PE<span class="sc">-</span><span class="cf">for</span><span class="sc">-</span>idba.fa <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out <span class="sc">--</span>mink <span class="dv">20</span> <span class="sc">--</span>maxk <span class="dv">50</span> <span class="sc">--</span>step <span class="dv">5</span> <span class="sc">--</span>num_threads <span class="dv">16</span></span>
<span id="cb10-12"><a href="#cb10-12"></a></span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="co"># Make sure fastq files are re-zipped:</span></span>
<span id="cb10-14"><a href="#cb10-14"></a>gzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq</span>
<span id="cb10-15"><a href="#cb10-15"></a>gzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Using 8 cores, the IDBA-tran assembly took about 23 hours and used 130 GB of memory.</p>
<p>The <em>IDBA-tran output</em> includes separation by k-mer length. So there are multiple “contig-X.fa” files. Where <code>contig-25.fa</code> is the output contigs at k-mer = 25. The IDBA program determines which one is best and then puts this into the final <code>contig.fa</code> file. Output: <code>/scratch/group/hu-lab/meta-eukomics/assembly/idba-out/contig.fa</code></p>
</section>
</section>
<section id="evaluate-assemblies" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="evaluate-assemblies"><span class="header-section-number">2.5</span> 5. Evaluate assemblies</h2>
<section id="quast" class="level3">
<h3 class="anchored" data-anchor-id="quast">QUAST</h3>
<p>QUAST stands for <em>QUality ASsessment Tool</em> and it is a tool for evaluating assemblies. Not all tools for evaluating assemblies will work, as most are built for metagenomics or require a reference genome. In this case, we do not have a reference genome.</p>
<p><em>First for the IDBA output</em>:</p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>module load GCC<span class="sc">/</span><span class="dv">9</span>.<span class="fl">3.0</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>module load OpenMPI<span class="sc">/</span><span class="dv">4</span>.<span class="fl">0.3</span></span>
<span id="cb11-3"><a href="#cb11-3"></a>module load QUAST<span class="sc">/</span><span class="dv">5</span>.<span class="fl">0.2</span><span class="sc">-</span>Python<span class="dv">-3</span>.<span class="fl">8.2</span></span>
<span id="cb11-4"><a href="#cb11-4"></a></span>
<span id="cb11-5"><a href="#cb11-5"></a>quast.py <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out<span class="sc">/</span>contig.fa \</span>
<span id="cb11-6"><a href="#cb11-6"></a>        <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb11-7"><a href="#cb11-7"></a>        <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out<span class="sc">/</span> \</span>
<span id="cb11-8"><a href="#cb11-8"></a>        <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This was run on 1 node with 16 cores. It used 3.5 GB of memory in about 1 hour. 6:15 CPUs used.</p>
<p><em>Repeat for the megahit output</em></p>
<div class="cell">
<details open="">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>module load GCC<span class="sc">/</span><span class="dv">9</span>.<span class="fl">3.0</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>module load OpenMPI<span class="sc">/</span><span class="dv">4</span>.<span class="fl">0.3</span></span>
<span id="cb12-3"><a href="#cb12-3"></a>module load QUAST<span class="sc">/</span><span class="dv">5</span>.<span class="fl">0.2</span><span class="sc">-</span>Python<span class="dv">-3</span>.<span class="fl">8.2</span></span>
<span id="cb12-4"><a href="#cb12-4"></a></span>
<span id="cb12-5"><a href="#cb12-5"></a>quast.py <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>megahit<span class="sc">-</span>output<span class="sc">/</span>final.contigs.fa \</span>
<span id="cb12-6"><a href="#cb12-6"></a>        <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb12-7"><a href="#cb12-7"></a>        <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>megahit<span class="sc">-</span>output<span class="sc">/</span> \</span>
<span id="cb12-8"><a href="#cb12-8"></a>        <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="interpret-quast" class="level4">
<h4 class="anchored" data-anchor-id="interpret-quast">Interpret QUAST</h4>
<p>Download files locally from:</p>
<ul>
<li>PDF files: <code>meta-eukomics/assembly/*-out/basic_stats/*.pdf</code></li>
<li>Use an ICARUS html viewer: <code>meta-eukomics/assembly/*-out/icarus*</code></li>
<li>Basic read stats: <code>meta-eukomics/assembly/*-out/reads_stats/reads_report.tsv</code></li>
</ul>
</section>
</section>
<section id="section" class="level3">
<h3 class="anchored" data-anchor-id="section"></h3>
</section>
</section>
<section id="metatranscriptome-run-information" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="metatranscriptome-run-information"><span class="header-section-number">2.6</span> Metatranscriptome run information</h2>
<table class="table">
<thead>
<tr class="header">
<th>Software</th>
<th>Version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fastqc</td>
<td>0.11.9</td>
</tr>
<tr class="even">
<td>Trimmomatic</td>
<td>0.39</td>
</tr>
<tr class="odd">
<td>SortMeRNA</td>
<td>4.3.7</td>
</tr>
<tr class="even">
<td>megahit</td>
<td>1.2.9</td>
</tr>
<tr class="odd">
<td>IDBA-Tran</td>
<td>1.1.3</td>
</tr>
<tr class="even">
<td>quast</td>
<td>5.0.2</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
</tr>
</tbody>
</table>
<section id="citations" class="level4">
<h4 class="anchored" data-anchor-id="citations">Citations</h4>
<ul>
<li><p>[fastqc]</p></li>
<li><p>[trimmomatic]</p></li>
<li><p>SortMeRNA: Kopylova E., Noe L. and Touzet H., “SortMeRNA: Fast and accurate filtering of ribosomal RNAs in metatranscriptomic data”, Bioinformatics (2012), doi: 10.1093/bioinformatics/bts611.</p></li>
<li><p>[megahit]</p></li>
<li><p><a href="https://academic.oup.com/bioinformatics/article/29/13/i326/191893">IDBA-Tran</a></p></li>
</ul>
</section>
</section>
</section>
<section id="eukrhythmic" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> eukrhythmic</h1>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb13" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb13-1"><a href="#cb13-1"></a><span class="co">---</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="an">title:</span><span class="co"> "Meta-eukomic"</span></span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="an">format:</span></span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co">  html:</span></span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="co">    toc: true</span></span>
<span id="cb13-6"><a href="#cb13-6"></a><span class="co">    toc-location: left</span></span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="co">    number-sections: true</span></span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="co">    number-depth: 2</span></span>
<span id="cb13-9"><a href="#cb13-9"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb13-10"><a href="#cb13-10"></a><span class="co">---</span></span>
<span id="cb13-11"><a href="#cb13-11"></a></span>
<span id="cb13-12"><a href="#cb13-12"></a><span class="fu"># Meta-eukomic</span></span>
<span id="cb13-13"><a href="#cb13-13"></a></span>
<span id="cb13-14"><a href="#cb13-14"></a>*SKH* last updated June 2024</span>
<span id="cb13-15"><a href="#cb13-15"></a></span>
<span id="cb13-16"><a href="#cb13-16"></a>Location for raw files: <span class="in">`/scratch/group/hu-lab/meta-eukomics/raw-data`</span> Individual R1 and R2 files are 2.9 and 2.7 G, respectively.</span>
<span id="cb13-17"><a href="#cb13-17"></a></span>
<span id="cb13-18"><a href="#cb13-18"></a><span class="fu">## Background reading</span></span>
<span id="cb13-19"><a href="#cb13-19"></a></span>
<span id="cb13-20"><a href="#cb13-20"></a><span class="ss">1.  </span>Cohen NR, Alexander H, Krinos AI, Hu SK, Lampe RH. Marine Microeukaryote Metatranscriptomics: Sample Processing and Bioinformatic Workflow Recommendations for Ecological Applications. Frontiers in Marine Science 2022; 9.</span>
<span id="cb13-21"><a href="#cb13-21"></a></span>
<span id="cb13-22"><a href="#cb13-22"></a><span class="ss">2.  </span>Krinos AI, Cohen NR, Follows MJ, Alexander H. Reverse engineering environmental metatranscriptomes clarifies best practices for eukaryotic assembly. BMC Bioinformatics 2023; 24: 74.</span>
<span id="cb13-23"><a href="#cb13-23"></a></span>
<span id="cb13-24"><a href="#cb13-24"></a><span class="fu">## Working environment</span></span>
<span id="cb13-25"><a href="#cb13-25"></a></span>
<span id="cb13-26"><a href="#cb13-26"></a>High performace computer (HPC) hosted by my University. This is the TAMU HPRC</span>
<span id="cb13-27"><a href="#cb13-27"></a></span>
<span id="cb13-28"><a href="#cb13-28"></a><span class="fu"># Step-by-step</span></span>
<span id="cb13-29"><a href="#cb13-29"></a></span>
<span id="cb13-30"><a href="#cb13-30"></a><span class="fu">## 1. Determine assembly groups</span></span>
<span id="cb13-31"><a href="#cb13-31"></a></span>
<span id="cb13-32"><a href="#cb13-32"></a><span class="fu">## 2. Trim &amp; QC</span></span>
<span id="cb13-33"><a href="#cb13-33"></a></span>
<span id="cb13-34"><a href="#cb13-34"></a>*Initial Fastqc*</span>
<span id="cb13-35"><a href="#cb13-35"></a></span>
<span id="cb13-36"><a href="#cb13-36"></a>Code for slurm script run on the HPC to run fastqc (<span class="in">`fastqc.slurm`</span>). For this set of samples, it took 10 minutes and used 290 MB. CPU used: 00:12:27.</span>
<span id="cb13-37"><a href="#cb13-37"></a></span>
<span id="cb13-40"><a href="#cb13-40"></a><span class="in">```{r}</span></span>
<span id="cb13-41"><a href="#cb13-41"></a>module load FastQC<span class="sc">/</span><span class="dv">0</span>.<span class="fl">11.9</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb13-42"><a href="#cb13-42"></a></span>
<span id="cb13-43"><a href="#cb13-43"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001.fastq.gz</span>
<span id="cb13-44"><a href="#cb13-44"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_001.fastq.gz</span>
<span id="cb13-45"><a href="#cb13-45"></a><span class="in">```</span></span>
<span id="cb13-46"><a href="#cb13-46"></a></span>
<span id="cb13-47"><a href="#cb13-47"></a></span>
<span id="cb13-48"><a href="#cb13-48"></a>In order to look at the output <span class="in">`.html`</span> files, they need to be opened locally.</span>
<span id="cb13-49"><a href="#cb13-49"></a></span>
<span id="cb13-52"><a href="#cb13-52"></a><span class="in">```{r}</span></span>
<span id="cb13-53"><a href="#cb13-53"></a>scp <span class="sc">$</span>HPRC<span class="sc">-</span>ADDRESS<span class="sc">:</span><span class="er">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001_fastqc.html meta<span class="sc">-</span>eukomic<span class="sc">/</span>output<span class="sc">-</span>files<span class="sc">/</span></span>
<span id="cb13-54"><a href="#cb13-54"></a><span class="in">```</span></span>
<span id="cb13-55"><a href="#cb13-55"></a></span>
<span id="cb13-56"><a href="#cb13-56"></a>Then we need to trim the individual reads, removing any sequencing-based primers, etc. We can use the program, <span class="co">[</span><span class="ot">Trimmomatic</span><span class="co">](http://www.usadellab.org/cms/?page=trimmomatic)</span> for this. R1 reads are forward and R2 are reverse. The Trimmomatic software requires input reads and then output trimmed and unpaired reads (the latter of which are discarded). Another input file required is a list of the possible primers and adapters from sequencing, <span class="in">`adapters-primers.fa`</span>.</span>
<span id="cb13-57"><a href="#cb13-57"></a></span>
<span id="cb13-58"><a href="#cb13-58"></a>Slurm script, <span class="in">`trim_fastqc.slurm`</span>. Below, trim parameter include:</span>
<span id="cb13-59"><a href="#cb13-59"></a></span>
<span id="cb13-60"><a href="#cb13-60"></a><span class="ss">* </span>Remove adapters, found in <span class="in">`adapters-primers.fa`</span></span>
<span id="cb13-61"><a href="#cb13-61"></a></span>
<span id="cb13-62"><a href="#cb13-62"></a><span class="ss">* </span>Remove leading low quality or N bases (below quality 3) (LEADING:3)</span>
<span id="cb13-63"><a href="#cb13-63"></a></span>
<span id="cb13-64"><a href="#cb13-64"></a><span class="ss">* </span>Remove trailing low quality or N bases (below quality 3) (TRAILING:3)</span>
<span id="cb13-65"><a href="#cb13-65"></a></span>
<span id="cb13-66"><a href="#cb13-66"></a><span class="ss">* </span>Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 10 (SLIDINGWINDOW:4:10)</span>
<span id="cb13-67"><a href="#cb13-67"></a></span>
<span id="cb13-68"><a href="#cb13-68"></a><span class="ss">* </span>Drop reads shorter than 50 bases long (MINLEN:50)</span>
<span id="cb13-69"><a href="#cb13-69"></a></span>
<span id="cb13-72"><a href="#cb13-72"></a><span class="in">```{r}</span></span>
<span id="cb13-73"><a href="#cb13-73"></a>module load Trimmomatic<span class="sc">/</span><span class="fl">0.39</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb13-74"><a href="#cb13-74"></a></span>
<span id="cb13-75"><a href="#cb13-75"></a>java <span class="sc">-</span>jar <span class="sc">$</span>EBROOTTRIMMOMATIC<span class="sc">/</span>trimmomatic<span class="dv">-0</span>.<span class="fl">39.</span>jar PE <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_001.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_001.fastq.gz  <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_unpaired.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_unpaired.fastq.gz ILLUMINACLIP<span class="sc">:</span>adapters<span class="sc">-</span>primers.fa<span class="sc">:</span><span class="dv">2</span><span class="sc">:</span><span class="dv">30</span><span class="sc">:</span><span class="dv">10</span> LEADING<span class="sc">:</span><span class="dv">3</span> TRAILING<span class="sc">:</span><span class="dv">3</span> SLIDINGWINDOW<span class="sc">:</span><span class="dv">4</span><span class="sc">:</span><span class="dv">10</span> MINLEN<span class="sc">:</span><span class="dv">50</span></span>
<span id="cb13-76"><a href="#cb13-76"></a></span>
<span id="cb13-77"><a href="#cb13-77"></a><span class="co"># "$EBROOTTRIMMOMATIC/trimmomatic-0.39.jar" note that this is specific to the HPRC system we are using</span></span>
<span id="cb13-78"><a href="#cb13-78"></a></span>
<span id="cb13-79"><a href="#cb13-79"></a>echo <span class="st">"Trimmomtatic complete. Repeating fastqc"</span></span>
<span id="cb13-80"><a href="#cb13-80"></a></span>
<span id="cb13-81"><a href="#cb13-81"></a>module load FastQC<span class="sc">/</span><span class="dv">0</span>.<span class="fl">11.9</span><span class="sc">-</span>Java<span class="dv">-11</span></span>
<span id="cb13-82"><a href="#cb13-82"></a></span>
<span id="cb13-83"><a href="#cb13-83"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz</span>
<span id="cb13-84"><a href="#cb13-84"></a>fastqc <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz</span>
<span id="cb13-85"><a href="#cb13-85"></a><span class="in">```</span></span>
<span id="cb13-86"><a href="#cb13-86"></a></span>
<span id="cb13-87"><a href="#cb13-87"></a><span class="fu">## 3. Remove ribosomal RNA</span></span>
<span id="cb13-88"><a href="#cb13-88"></a></span>
<span id="cb13-89"><a href="#cb13-89"></a><span class="co">[</span><span class="ot">Sortmerna</span><span class="co">](https://sortmerna.readthedocs.io/en/latest/installation.html)</span> installation guidelines and usage can be found here.</span>
<span id="cb13-90"><a href="#cb13-90"></a></span>
<span id="cb13-91"><a href="#cb13-91"></a>On the TAMU HPRC, this is where we needed to make a conda environment to run this.</span>
<span id="cb13-92"><a href="#cb13-92"></a></span>
<span id="cb13-95"><a href="#cb13-95"></a><span class="in">```{r}</span></span>
<span id="cb13-96"><a href="#cb13-96"></a><span class="co"># Load anaconda</span></span>
<span id="cb13-97"><a href="#cb13-97"></a>module load Anaconda3<span class="sc">/</span><span class="fl">2022.05</span></span>
<span id="cb13-98"><a href="#cb13-98"></a></span>
<span id="cb13-99"><a href="#cb13-99"></a><span class="co"># modify bioconda config for sortmeRNA</span></span>
<span id="cb13-100"><a href="#cb13-100"></a>conda config <span class="sc">--</span>add channels defaults</span>
<span id="cb13-101"><a href="#cb13-101"></a>conda config <span class="sc">--</span>add channels bioconda</span>
<span id="cb13-102"><a href="#cb13-102"></a>conda config <span class="sc">--</span>add channels conda<span class="sc">-</span>forge</span>
<span id="cb13-103"><a href="#cb13-103"></a>conda config <span class="sc">--</span>set channel_priority strict</span>
<span id="cb13-104"><a href="#cb13-104"></a><span class="in">```</span></span>
<span id="cb13-105"><a href="#cb13-105"></a></span>
<span id="cb13-106"><a href="#cb13-106"></a>Search for sortmerna using conda:</span>
<span id="cb13-109"><a href="#cb13-109"></a><span class="in">```{r}</span></span>
<span id="cb13-110"><a href="#cb13-110"></a>conda search sortmerna</span>
<span id="cb13-111"><a href="#cb13-111"></a><span class="in">```</span></span>
<span id="cb13-112"><a href="#cb13-112"></a>Create a conda environment specific for sortmerna, activate it, and install it.</span>
<span id="cb13-115"><a href="#cb13-115"></a><span class="in">```{r}</span></span>
<span id="cb13-116"><a href="#cb13-116"></a>conda create <span class="sc">--</span>name sortmerna_env</span>
<span id="cb13-117"><a href="#cb13-117"></a>conda activate sortmerna_env</span>
<span id="cb13-118"><a href="#cb13-118"></a>conda install sortmerna</span>
<span id="cb13-119"><a href="#cb13-119"></a><span class="in">```</span></span>
<span id="cb13-120"><a href="#cb13-120"></a></span>
<span id="cb13-121"><a href="#cb13-121"></a>Answer "yes" (*Y*) to installation questions. Now when you run SortMeRNA, you need to activate this conda environment and execute your code. The next step is to download the databases to use for this program.</span>
<span id="cb13-122"><a href="#cb13-122"></a></span>
<span id="cb13-123"><a href="#cb13-123"></a></span>
<span id="cb13-124"><a href="#cb13-124"></a><span class="fu">#### SortMeRNA databases</span></span>
<span id="cb13-125"><a href="#cb13-125"></a></span>
<span id="cb13-126"><a href="#cb13-126"></a>Separately, you need to download the databases to "align" and check the sequences for ribosomal RNAs.</span>
<span id="cb13-127"><a href="#cb13-127"></a></span>
<span id="cb13-128"><a href="#cb13-128"></a></span>
<span id="cb13-129"><a href="#cb13-129"></a>Instructions for <span class="co">[</span><span class="ot">downloading version 4</span><span class="co">](https://sortmerna.readthedocs.io/en/latest/databases.html)</span>:</span>
<span id="cb13-132"><a href="#cb13-132"></a><span class="in">```{r}</span></span>
<span id="cb13-133"><a href="#cb13-133"></a><span class="co"># Move to the location you want your rRNA databases stored</span></span>
<span id="cb13-134"><a href="#cb13-134"></a>cd <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span> </span>
<span id="cb13-135"><a href="#cb13-135"></a></span>
<span id="cb13-136"><a href="#cb13-136"></a><span class="co"># Download from github release </span></span>
<span id="cb13-137"><a href="#cb13-137"></a>wget https<span class="sc">:</span><span class="er">//</span>github.com<span class="sc">/</span>biocore<span class="sc">/</span>sortmerna<span class="sc">/</span>releases<span class="sc">/</span>download<span class="sc">/</span>v4.<span class="fl">3.4</span><span class="sc">/</span>database.tar.gz</span>
<span id="cb13-138"><a href="#cb13-138"></a></span>
<span id="cb13-139"><a href="#cb13-139"></a><span class="co"># Make new directory for this and un-zip downloaded file</span></span>
<span id="cb13-140"><a href="#cb13-140"></a>mkdir rRNA_databases_v4</span>
<span id="cb13-141"><a href="#cb13-141"></a>tar <span class="sc">-</span>xvf database.tar.gz <span class="sc">-</span>C rRNA_databases_v4</span>
<span id="cb13-142"><a href="#cb13-142"></a><span class="in">```</span></span>
<span id="cb13-143"><a href="#cb13-143"></a></span>
<span id="cb13-144"><a href="#cb13-144"></a><span class="fu">### Run SortMeRNA</span></span>
<span id="cb13-145"><a href="#cb13-145"></a></span>
<span id="cb13-146"><a href="#cb13-146"></a></span>
<span id="cb13-147"><a href="#cb13-147"></a>In your slurm script, activate conda, and then the sortmerna environment:</span>
<span id="cb13-150"><a href="#cb13-150"></a><span class="in">```{r}</span></span>
<span id="cb13-151"><a href="#cb13-151"></a><span class="co"># Load anaconda</span></span>
<span id="cb13-152"><a href="#cb13-152"></a>module load Anaconda3<span class="sc">/</span><span class="fl">2022.05</span></span>
<span id="cb13-153"><a href="#cb13-153"></a>conda activate sortmerna_env</span>
<span id="cb13-154"><a href="#cb13-154"></a></span>
<span id="cb13-155"><a href="#cb13-155"></a>mkdir <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA<span class="sc">-</span>sort</span>
<span id="cb13-156"><a href="#cb13-156"></a></span>
<span id="cb13-157"><a href="#cb13-157"></a>sortmerna <span class="sc">-</span>ref <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA_databases_v4<span class="sc">/</span>smr_v4<span class="fl">.3</span>_sensitive_db.fasta \</span>
<span id="cb13-158"><a href="#cb13-158"></a><span class="sc">-</span>reads <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz \</span>
<span id="cb13-159"><a href="#cb13-159"></a><span class="sc">-</span>reads <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb13-160"><a href="#cb13-160"></a><span class="sc">-</span>workdir <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>rRNA<span class="sc">-</span>sort<span class="sc">/</span></span>
<span id="cb13-161"><a href="#cb13-161"></a><span class="sc">-</span>fastx <span class="sc">--</span>paired_in <span class="sc">--</span>out2 <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span>
<span id="cb13-162"><a href="#cb13-162"></a><span class="in">```</span></span>
<span id="cb13-163"><a href="#cb13-163"></a></span>
<span id="cb13-164"><a href="#cb13-164"></a><span class="in">`paired_in`</span>: The option ‘paired_in’ is optimal for users who want all reads in the other.fasta file to be non-rRNA. However, there are small chances that reads which are non-rRNA will also be put into the aligned.fasta file.</span>
<span id="cb13-165"><a href="#cb13-165"></a></span>
<span id="cb13-166"><a href="#cb13-166"></a>Since the SortmeRNA tool is optimitized to take things that are aligned with the rRNA databases and put them in a file called "aligned", then we need to use the <span class="in">`other.fasta`</span> output file. </span>
<span id="cb13-167"><a href="#cb13-167"></a></span>
<span id="cb13-168"><a href="#cb13-168"></a>How do I properly run sortmerna? https://github.com/sortmerna/sortmerna/issues/333</span>
<span id="cb13-169"><a href="#cb13-169"></a></span>
<span id="cb13-170"><a href="#cb13-170"></a></span>
<span id="cb13-171"><a href="#cb13-171"></a></span>
<span id="cb13-172"><a href="#cb13-172"></a><span class="fu">## 4. Assembly</span></span>
<span id="cb13-173"><a href="#cb13-173"></a></span>
<span id="cb13-174"><a href="#cb13-174"></a>The next step is to take all the sequenced reads and bring them together to make longer, more continuous sequences, called *contigs*. Here, we will use two assemblers and combine the results. Each one is built slightly differently.</span>
<span id="cb13-175"><a href="#cb13-175"></a></span>
<span id="cb13-176"><a href="#cb13-176"></a><span class="fu">### MEGAHIT (assembly 1)</span></span>
<span id="cb13-177"><a href="#cb13-177"></a></span>
<span id="cb13-178"><a href="#cb13-178"></a>We will first use <span class="co">[</span><span class="ot">megahit</span><span class="co">](https://github.com/voutcn/megahit)</span>. To save the assemblies separately, make a new assembly output file in your working <span class="in">`scratch`</span> space. <span class="in">`mkdir /scratch/group/hu-lab/meta-eukomics/assembly`</span></span>
<span id="cb13-179"><a href="#cb13-179"></a></span>
<span id="cb13-180"><a href="#cb13-180"></a>The megahit command below, outputs contigs in the new <span class="in">`assembly`</span> directory, only keeps reads longer than 100 bps, and uses the megahit preset for lots of diversity in a sample (_meta-large_).</span>
<span id="cb13-181"><a href="#cb13-181"></a></span>
<span id="cb13-182"><a href="#cb13-182"></a>Megahit uses multiple k-mer strategy, and we can set the min and max k. In order to reduce the complexity of the _de Bruijin_ graph, a kmin size of 25 and higher (like 27) is better. </span>
<span id="cb13-183"><a href="#cb13-183"></a></span>
<span id="cb13-184"><a href="#cb13-184"></a></span>
<span id="cb13-185"><a href="#cb13-185"></a>See slurm script: <span class="in">`megahit-assembly.slurm`</span></span>
<span id="cb13-188"><a href="#cb13-188"></a><span class="in">```{r}</span></span>
<span id="cb13-189"><a href="#cb13-189"></a>module load GCCcore<span class="sc">/</span><span class="dv">11</span>.<span class="fl">2.0</span></span>
<span id="cb13-190"><a href="#cb13-190"></a>module load MEGAHIT<span class="sc">/</span><span class="dv">1</span>.<span class="fl">2.9</span></span>
<span id="cb13-191"><a href="#cb13-191"></a></span>
<span id="cb13-192"><a href="#cb13-192"></a>megahit <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span> <span class="sc">--</span>min<span class="sc">-</span>contig<span class="sc">-</span>len <span class="dv">1000</span> <span class="sc">--</span>presets meta<span class="sc">-</span>large <span class="sc">--</span>num<span class="sc">-</span>cpu<span class="sc">-</span>threads <span class="dv">32</span></span>
<span id="cb13-193"><a href="#cb13-193"></a><span class="in">```</span></span>
<span id="cb13-194"><a href="#cb13-194"></a></span>
<span id="cb13-195"><a href="#cb13-195"></a>With the above settings, megahit recovered 31,331 contigs, total 44610184 bp, min 1000 bp, max 10674 bp, avg 1423 bp, N50 1391 bp. And this took about *10 hours*.</span>
<span id="cb13-196"><a href="#cb13-196"></a></span>
<span id="cb13-197"><a href="#cb13-197"></a>On the HPC, this was run on 1 node, 4 cores per node, and it used 18.5 GBs of memory.</span>
<span id="cb13-198"><a href="#cb13-198"></a></span>
<span id="cb13-199"><a href="#cb13-199"></a></span>
<span id="cb13-200"><a href="#cb13-200"></a>Megahit assembly location: <span class="in">`/scratch/group/hu-lab/meta-eukomics/assembly/megahit-output/final.contigs.fa`</span></span>
<span id="cb13-201"><a href="#cb13-201"></a></span>
<span id="cb13-202"><a href="#cb13-202"></a></span>
<span id="cb13-203"><a href="#cb13-203"></a>*_Optional_: Visualize your megahit assembly*</span>
<span id="cb13-204"><a href="#cb13-204"></a></span>
<span id="cb13-205"><a href="#cb13-205"></a>When megahit assembly is complete, use this option to visualize it: https://github.com/voutcn/megahit/wiki/Visualizing-MEGAHIT's-contig-graph</span>
<span id="cb13-206"><a href="#cb13-206"></a></span>
<span id="cb13-207"><a href="#cb13-207"></a></span>
<span id="cb13-208"><a href="#cb13-208"></a><span class="fu">### IDBA-Tran</span></span>
<span id="cb13-209"><a href="#cb13-209"></a></span>
<span id="cb13-210"><a href="#cb13-210"></a><span class="co">[</span><span class="ot">IDBA-tran</span><span class="co">](https://i.cs.hku.hk/~alse/hkubrg/projects/idba_tran/index.html)</span> is another _de novo_ assembler. Uses local assembly to reconstruct missing kmers in low-expressed transcripts.</span>
<span id="cb13-211"><a href="#cb13-211"></a></span>
<span id="cb13-212"><a href="#cb13-212"></a>See script: <span class="in">`idba-assembly.slurm`</span> to run assembly with minimum kmer at 20 and max kmer at 50 with a 5 step increment of kmer.</span>
<span id="cb13-215"><a href="#cb13-215"></a><span class="in">```{r}</span></span>
<span id="cb13-216"><a href="#cb13-216"></a>echo <span class="st">"unzip files"</span></span>
<span id="cb13-217"><a href="#cb13-217"></a>gunzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R<span class="sc">*</span>_trimmed.fastq.gz</span>
<span id="cb13-218"><a href="#cb13-218"></a></span>
<span id="cb13-219"><a href="#cb13-219"></a>module load GCC<span class="sc">/</span><span class="dv">8</span>.<span class="fl">2.0</span><span class="dv">-2</span>.<span class="fl">31.1</span> </span>
<span id="cb13-220"><a href="#cb13-220"></a>module load IDBA<span class="sc">-</span>UD<span class="sc">/</span><span class="dv">1</span>.<span class="fl">1.3</span></span>
<span id="cb13-221"><a href="#cb13-221"></a></span>
<span id="cb13-222"><a href="#cb13-222"></a><span class="co"># Convert fastq R1 and R2 files into a single 'interleaved` fastq files</span></span>
<span id="cb13-223"><a href="#cb13-223"></a>fq2fa <span class="sc">--</span>merge <span class="sc">--</span>filter <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>merged<span class="sc">-</span>PE<span class="sc">-</span><span class="cf">for</span><span class="sc">-</span>idba.fa</span>
<span id="cb13-224"><a href="#cb13-224"></a></span>
<span id="cb13-225"><a href="#cb13-225"></a><span class="co"># Run assembly</span></span>
<span id="cb13-226"><a href="#cb13-226"></a>idba_tran <span class="sc">-</span>r <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>merged<span class="sc">-</span>PE<span class="sc">-</span><span class="cf">for</span><span class="sc">-</span>idba.fa <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out <span class="sc">--</span>mink <span class="dv">20</span> <span class="sc">--</span>maxk <span class="dv">50</span> <span class="sc">--</span>step <span class="dv">5</span> <span class="sc">--</span>num_threads <span class="dv">16</span></span>
<span id="cb13-227"><a href="#cb13-227"></a></span>
<span id="cb13-228"><a href="#cb13-228"></a><span class="co"># Make sure fastq files are re-zipped:</span></span>
<span id="cb13-229"><a href="#cb13-229"></a>gzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq</span>
<span id="cb13-230"><a href="#cb13-230"></a>gzip <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq</span>
<span id="cb13-231"><a href="#cb13-231"></a><span class="in">```</span></span>
<span id="cb13-232"><a href="#cb13-232"></a></span>
<span id="cb13-233"><a href="#cb13-233"></a>Using 8 cores, the IDBA-tran assembly took about 23 hours and used 130 GB of memory.</span>
<span id="cb13-234"><a href="#cb13-234"></a></span>
<span id="cb13-235"><a href="#cb13-235"></a></span>
<span id="cb13-236"><a href="#cb13-236"></a>The *IDBA-tran output* includes separation by k-mer length. So there are multiple "contig-X.fa" files. Where <span class="in">`contig-25.fa`</span> is the output contigs at k-mer = 25. The IDBA program determines which one is best and then puts this into the final <span class="in">`contig.fa`</span> file. </span>
<span id="cb13-237"><a href="#cb13-237"></a>Output: <span class="in">`/scratch/group/hu-lab/meta-eukomics/assembly/idba-out/contig.fa`</span></span>
<span id="cb13-238"><a href="#cb13-238"></a></span>
<span id="cb13-239"><a href="#cb13-239"></a><span class="fu">## 5. Evaluate assemblies</span></span>
<span id="cb13-240"><a href="#cb13-240"></a></span>
<span id="cb13-241"><a href="#cb13-241"></a><span class="fu">### QUAST</span></span>
<span id="cb13-242"><a href="#cb13-242"></a></span>
<span id="cb13-243"><a href="#cb13-243"></a>QUAST stands for _QUality ASsessment Tool_ and it is a tool for evaluating assemblies. Not all tools for evaluating assemblies will work, as most are built for metagenomics or require a reference genome. In this case, we do not have a reference genome. </span>
<span id="cb13-244"><a href="#cb13-244"></a></span>
<span id="cb13-245"><a href="#cb13-245"></a></span>
<span id="cb13-246"><a href="#cb13-246"></a>*First for the IDBA output*:</span>
<span id="cb13-247"><a href="#cb13-247"></a></span>
<span id="cb13-250"><a href="#cb13-250"></a><span class="in">```{r}</span></span>
<span id="cb13-251"><a href="#cb13-251"></a>module load GCC<span class="sc">/</span><span class="dv">9</span>.<span class="fl">3.0</span></span>
<span id="cb13-252"><a href="#cb13-252"></a>module load OpenMPI<span class="sc">/</span><span class="dv">4</span>.<span class="fl">0.3</span></span>
<span id="cb13-253"><a href="#cb13-253"></a>module load QUAST<span class="sc">/</span><span class="dv">5</span>.<span class="fl">0.2</span><span class="sc">-</span>Python<span class="dv">-3</span>.<span class="fl">8.2</span></span>
<span id="cb13-254"><a href="#cb13-254"></a></span>
<span id="cb13-255"><a href="#cb13-255"></a>quast.py <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out<span class="sc">/</span>contig.fa \</span>
<span id="cb13-256"><a href="#cb13-256"></a>        <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb13-257"><a href="#cb13-257"></a>        <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>idba<span class="sc">-</span>out<span class="sc">/</span> \</span>
<span id="cb13-258"><a href="#cb13-258"></a>        <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span>
<span id="cb13-259"><a href="#cb13-259"></a><span class="in">```</span></span>
<span id="cb13-260"><a href="#cb13-260"></a></span>
<span id="cb13-261"><a href="#cb13-261"></a>This was run on 1 node with 16 cores. It used 3.5 GB of memory in about 1 hour. 6:15 CPUs used.</span>
<span id="cb13-262"><a href="#cb13-262"></a></span>
<span id="cb13-263"><a href="#cb13-263"></a></span>
<span id="cb13-264"><a href="#cb13-264"></a></span>
<span id="cb13-265"><a href="#cb13-265"></a>*Repeat for the megahit output*</span>
<span id="cb13-266"><a href="#cb13-266"></a></span>
<span id="cb13-269"><a href="#cb13-269"></a><span class="in">```{r}</span></span>
<span id="cb13-270"><a href="#cb13-270"></a>module load GCC<span class="sc">/</span><span class="dv">9</span>.<span class="fl">3.0</span></span>
<span id="cb13-271"><a href="#cb13-271"></a>module load OpenMPI<span class="sc">/</span><span class="dv">4</span>.<span class="fl">0.3</span></span>
<span id="cb13-272"><a href="#cb13-272"></a>module load QUAST<span class="sc">/</span><span class="dv">5</span>.<span class="fl">0.2</span><span class="sc">-</span>Python<span class="dv">-3</span>.<span class="fl">8.2</span></span>
<span id="cb13-273"><a href="#cb13-273"></a></span>
<span id="cb13-274"><a href="#cb13-274"></a>quast.py <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>megahit<span class="sc">-</span>output<span class="sc">/</span>final.contigs.fa \</span>
<span id="cb13-275"><a href="#cb13-275"></a>        <span class="sc">-</span><span class="dv">1</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R1_trimmed.fastq.gz <span class="sc">-</span><span class="dv">2</span> <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>raw<span class="sc">-</span>data<span class="sc">/</span>HS039_S90_L004_R2_trimmed.fastq.gz \</span>
<span id="cb13-276"><a href="#cb13-276"></a>        <span class="sc">-</span>o <span class="sc">/</span>scratch<span class="sc">/</span>group<span class="sc">/</span>hu<span class="sc">-</span>lab<span class="sc">/</span>meta<span class="sc">-</span>eukomics<span class="sc">/</span>assembly<span class="sc">/</span>megahit<span class="sc">-</span>output<span class="sc">/</span> \</span>
<span id="cb13-277"><a href="#cb13-277"></a>        <span class="sc">--</span>threads <span class="sc">$</span>SLURM_CPUS_PER_TASK</span>
<span id="cb13-278"><a href="#cb13-278"></a><span class="in">```</span></span>
<span id="cb13-279"><a href="#cb13-279"></a></span>
<span id="cb13-280"><a href="#cb13-280"></a></span>
<span id="cb13-281"><a href="#cb13-281"></a><span class="fu">#### Interpret QUAST </span></span>
<span id="cb13-282"><a href="#cb13-282"></a></span>
<span id="cb13-283"><a href="#cb13-283"></a>Download files locally from:</span>
<span id="cb13-284"><a href="#cb13-284"></a></span>
<span id="cb13-285"><a href="#cb13-285"></a><span class="ss">  * </span>PDF files: <span class="in">`meta-eukomics/assembly/*-out/basic_stats/*.pdf `</span></span>
<span id="cb13-286"><a href="#cb13-286"></a><span class="ss">  * </span>Use an ICARUS html viewer: <span class="in">`meta-eukomics/assembly/*-out/icarus* `</span></span>
<span id="cb13-287"><a href="#cb13-287"></a><span class="ss">  * </span>Basic read stats: <span class="in">`meta-eukomics/assembly/*-out/reads_stats/reads_report.tsv`</span></span>
<span id="cb13-288"><a href="#cb13-288"></a></span>
<span id="cb13-289"><a href="#cb13-289"></a></span>
<span id="cb13-290"><a href="#cb13-290"></a>###</span>
<span id="cb13-291"><a href="#cb13-291"></a></span>
<span id="cb13-292"><a href="#cb13-292"></a></span>
<span id="cb13-293"><a href="#cb13-293"></a></span>
<span id="cb13-294"><a href="#cb13-294"></a></span>
<span id="cb13-295"><a href="#cb13-295"></a><span class="fu">## Metatranscriptome run information</span></span>
<span id="cb13-296"><a href="#cb13-296"></a></span>
<span id="cb13-297"><a href="#cb13-297"></a>| Software    | Version |</span>
<span id="cb13-298"><a href="#cb13-298"></a>|-------------|---------|</span>
<span id="cb13-299"><a href="#cb13-299"></a>| Fastqc      | 0.11.9  |</span>
<span id="cb13-300"><a href="#cb13-300"></a>| Trimmomatic | 0.39    |</span>
<span id="cb13-301"><a href="#cb13-301"></a>| SortMeRNA   | 4.3.7   |</span>
<span id="cb13-302"><a href="#cb13-302"></a>| megahit     | 1.2.9   |</span>
<span id="cb13-303"><a href="#cb13-303"></a>| IDBA-Tran   | 1.1.3   |</span>
<span id="cb13-304"><a href="#cb13-304"></a>| quast       | 5.0.2   |</span>
<span id="cb13-305"><a href="#cb13-305"></a>|             |         |</span>
<span id="cb13-306"><a href="#cb13-306"></a>|             |         |</span>
<span id="cb13-307"><a href="#cb13-307"></a>|             |         |</span>
<span id="cb13-308"><a href="#cb13-308"></a>|             |         |</span>
<span id="cb13-309"><a href="#cb13-309"></a>|             |         |</span>
<span id="cb13-310"><a href="#cb13-310"></a></span>
<span id="cb13-311"><a href="#cb13-311"></a><span class="fu">#### Citations</span></span>
<span id="cb13-312"><a href="#cb13-312"></a></span>
<span id="cb13-313"><a href="#cb13-313"></a><span class="ss">* </span><span class="co">[</span><span class="ot">fastqc</span><span class="co">]</span></span>
<span id="cb13-314"><a href="#cb13-314"></a></span>
<span id="cb13-315"><a href="#cb13-315"></a><span class="ss">* </span><span class="co">[</span><span class="ot">trimmomatic</span><span class="co">]</span></span>
<span id="cb13-316"><a href="#cb13-316"></a></span>
<span id="cb13-317"><a href="#cb13-317"></a><span class="ss">* </span>SortMeRNA: Kopylova E., Noe L. and Touzet H., “SortMeRNA: Fast and accurate filtering of ribosomal RNAs in metatranscriptomic data”, Bioinformatics (2012), doi: 10.1093/bioinformatics/bts611.</span>
<span id="cb13-318"><a href="#cb13-318"></a></span>
<span id="cb13-319"><a href="#cb13-319"></a><span class="ss">* </span><span class="co">[</span><span class="ot">megahit</span><span class="co">]</span></span>
<span id="cb13-320"><a href="#cb13-320"></a></span>
<span id="cb13-321"><a href="#cb13-321"></a><span class="ss">* </span><span class="co">[</span><span class="ot">IDBA-Tran</span><span class="co">](https://academic.oup.com/bioinformatics/article/29/13/i326/191893)</span></span>
<span id="cb13-322"><a href="#cb13-322"></a></span>
<span id="cb13-323"><a href="#cb13-323"></a></span>
<span id="cb13-324"><a href="#cb13-324"></a><span class="fu"># eukrhythmic</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>